src/app/server/JointFabricDatastorage.h

```diff
@@ -0,0 +1,198 @@
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <lib/core/CHIPPersistentStorageDelegate.h>
+#include <lib/core/CHIPVendorIdentifiers.hpp>
+#include <lib/core/NodeId.h>
+#include <vector>
+
+namespace chip {
+namespace app {
+
+/**
+ * A struct which extends the DatastoreNodeInformationEntry type with FriendlyName buffer reservation.
+ */
+struct GenericDatastoreNodeInformationEntry
+    : public Clusters::JointFabricDatastore::Structs::DatastoreNodeInformationEntryStruct::Type
+{
+    GenericDatastoreNodeInformationEntry(NodeId nodeId = 0, FabricIndex fabricId = 0,
+                                         Clusters::JointFabricDatastore::DatastoreStateEnum state =
+                                             Clusters::JointFabricDatastore::DatastoreStateEnum::kUnknownEnumValue,
+                                         Optional<CharSpan> label = NullOptional)
+    {
+        Set(nodeId, fabricId, state, label);
+    }
+
+    GenericDatastoreNodeInformationEntry(const GenericDatastoreNodeInformationEntry & op) { *this = op; }
+
+    GenericDatastoreNodeInformationEntry & operator=(const GenericDatastoreNodeInformationEntry & op)
+    {
+        Set(op.nodeID, op.fabricIndex, op.commissioningStatusEntry.state, MakeOptional(op.friendlyName));
+        return *this;
+    }
+
+    void Set(NodeId nodeId, FabricIndex fabricId, Clusters::JointFabricDatastore::DatastoreStateEnum state,
+             Optional<CharSpan> label = NullOptional)
+    {
+        SetFabricIndex(fabricId);
+        this->nodeID                         = nodeId;
+        this->commissioningStatusEntry.state = state;
+        Set(label);
+    }
+
+    void Set(Optional<CharSpan> label = NullOptional)
+    {
+        if (label.HasValue())
+        {
+            memset(mFriendlyNameBuffer, 0, sizeof(mFriendlyNameBuffer));
+            if (label.Value().size() > sizeof(mFriendlyNameBuffer))
+            {
+                memcpy(mFriendlyNameBuffer, label.Value().data(), sizeof(mFriendlyNameBuffer));
+                this->friendlyName = CharSpan(mFriendlyNameBuffer, sizeof(mFriendlyNameBuffer));
+            }
+            else
+            {
+                memcpy(mFriendlyNameBuffer, label.Value().data(), label.Value().size());
+                this->friendlyName = CharSpan(mFriendlyNameBuffer, label.Value().size());
+            }
+        }
+        else
+        {
+            this->friendlyName = CharSpan();
+        }
+    }
+
+private:
+    static constexpr size_t kFriendlyNameMaxSize = 32u;
+
+    char mFriendlyNameBuffer[kFriendlyNameMaxSize];
+};
+
+class JointFabricDatastorage
```

Comment:
I would replace all "storage" with "store", in the end in the spec. it's JointFabricDatastore, so I don't see any reason into making it JointFabricDatastorage into the SDK.

Reply:
Please make the change above.

Reply:
I’ve changed the class name to JointFabricDatastore in the latest commit.

Reply:
Addressed in the latest commit.